consensus/hotstuff/eventhandler/event_handler.go

```diff
@@ -293,76 +388,41 @@ func (e *EventHandler) startNewView() error {
 		//  Therefore, if this replica is the leader, the following code is a no-op.
 	}
 
-	// as a replica of the current view, find and process the block for the current view
-	proposals := e.forks.GetProposalsForView(curView)
-	if len(proposals) == 0 {
-		// if there is no block stored before for the current view, then exit and keep waiting
-		log.Debug().Msg("waiting for proposal from leader")
-		return nil
-	}
-
-	// when there are multiple block proposals, we will just pick the first one.
-	// forks is responsible for slashing double proposal behavior, and
-	// event handler is aware of double proposals, but picking any should work and
-	// won't hurt safety
-	block := proposals[0].Block
-
-	log.Debug().
-		Uint64("block_view", block.View).
-		Hex("block_id", block.BlockID[:]).
-		Uint64("parent_view", block.QC.View).
-		Hex("parent_id", block.QC.BlockID[:]).
-		Hex("signer", block.ProposerID[:]).
-		Msg("processing cached proposal from leader")
-
-	return e.processBlockForCurrentView(block)
+	return nil
 }
 
 // processBlockForCurrentView processes the block for the current view.
 // It is called AFTER the block has been stored or found in Forks
 // It checks whether to vote for this block.
-// It might trigger a view change to go to a different view, which might re-enter this function.
-func (e *EventHandler) processBlockForCurrentView(block *model.Block) error {
+// No errors are expected during normal operation.
+func (e *EventHandler) processBlockForCurrentView(proposal *model.Proposal) error {
 	// sanity check that block is really for the current view:
 	curView := e.paceMaker.CurView()
+	block := proposal.Block
 	if block.View != curView {
-		return fmt.Errorf("sanity check fails: block proposal's view does not match with curView, (blockView: %v, curView: %v)",
-			block.View, curView)
+		// ignore outdated proposals in case we have moved forward
+		return nil
 	}
 	// leader (node ID) for next view
 	nextLeader, err := e.committee.LeaderForView(curView + 1)
 	if err != nil {
```

Comment:
Explicitly handling expected sentinel, adding doc
```suggestion
	if errors.Is(err, model.ErrViewForUnknownEpoch) {
		// We are attempting process a block in an unknown epoch
		// This should never happen, because:
		// * the compliance layer ensures proposals are passed to the event loop strictly after their parent
		// * the protocol state ensures that, before incorporating the first block of an epoch E, 
		//    either E is known or we have triggered epoch fallback mode - in either case the epoch for the
		//    current epoch is known
		return fmt.Errorf("attempting to process a block for current view in unknown epoch: %v", err.Error())
	}
	if err != nil {
```